python/xorbits/datasets/backends/huggingface/loader.py

```diff
@@ -0,0 +1,162 @@
+    num_chunks: int = Int32Field("num_chunks")
+    chunk_index: int = Int32Field("chunk_index")
+    data_files = ListField("data_files")
+
+    def __call__(self):
+        from datasets import load_dataset_builder
+        from datasets.packaged_modules.folder_based_builder.folder_based_builder import (
+            FolderBasedBuilder,
+        )
+
+        builder_kwargs = self._get_kwargs(load_dataset_builder, self.hf_kwargs)
+        builder = load_dataset_builder(self.path, **builder_kwargs)
+        data_files = builder.config.data_files
+        # TODO(codingl2k1): support multiple splits
+        split = self.hf_kwargs["split"]
+        # TODO(codingl2k1): not pass dtypes if no to_dataframe() called.
+        if builder.info.features:
+            dtypes = builder.info.features.arrow_schema.empty_table().to_pandas().dtypes
+        else:
+            dtypes = None
+        if dtypes is not None and builder.info.splits:
+            shape = (
+                builder.info.splits[split].num_examples,
+                len(dtypes),
+            )
+        else:
+            shape = (np.nan, np.nan)
+        # TODO(codingl2k1): check data_files if can be supported
+        # e.g. the datasets mariosasko/test_imagefolder_with_metadata has multiple
+        # data files, but some of them are meta, so we can't parallel load it.
+        if not isinstance(builder, FolderBasedBuilder):
+            if data_files and len(data_files[split]) > 1:
+                data_files = list(data_files[split])
+            else:
+                data_files = None
+        else:
+            data_files = None
+        self.data_files = data_files
+        return self.new_tileable([], dtypes=dtypes, shape=shape)
+
+    @classmethod
+    def _get_kwargs(cls, obj, kwargs):
+        sig_builder = inspect.signature(obj)
+        return {
+            key: kwargs[key] for key in sig_builder.parameters.keys() if key in kwargs
+        }
+
+    @classmethod
+    def tile(cls, op: "HuggingfaceLoader"):
+        assert len(op.inputs) == 0
+
+        data_files = op.data_files
+        # Set op.data_files to None, we don't want every chunk op copy this field.
+        op.data_files = None
+
+        chunks = []
+        if data_files is not None:
+            ctx = get_context()
+            # TODO(codingl2k1): make expect worker binding stable for cache reuse.
+            all_bands = [b for b in ctx.get_worker_bands() if b[1].startswith("numa-")]
+            for index, (f, band) in enumerate(
+                zip(data_files, itertools.cycle(all_bands))
+            ):
+                chunk_op = op.copy().reset_key()
+                assert f, "Invalid data file from DatasetBuilder."
+                chunk_op.single_data_file = f
+                chunk_op.num_chunks = len(data_files)
+                chunk_op.chunk_index = index
+                chunk_op.expect_band = band
+                c = chunk_op.new_chunk(inputs=[], index=index)
```

Comment:
Is index a tuple of int here? And chunk_op.chunk_index is not needed.

Reply:
I am not sure if the index field can be empty. It's not a dataframe index.

Reply:
For dataset chunk, indexes can be (0,), (1,) ...

Reply:
Got it. Thanks.